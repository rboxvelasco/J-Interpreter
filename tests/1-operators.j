NB. The objective of this test is to check the mandatory features of G that are not checked in dummy.j.

NB. Flip operator modifier.
NB. It will be tested in more complicated conditions through the test.
6 - 3
6 -~ 3
6 -~~ 3
] ~ 1    NB. It should have no effect here.

NB. Concatenation of lists.
x =: 0 2
y =: 4 5

z =: x, y
zflip =: x,~ y
z
zflip

NB. Generator i.
i. 1 + 4
i. _3

NB. Indexed access to lists.
1 2 5 { i. _10

NB. Binary to Unary operators.
double =: +:
square =: *:
double 1 2 3
square 1 + 0 1 2

NB. Fold.
+ / 1 2 3
* / 2 3 5
, ~ / 1 2 3

NB. Unary # (size)
# 0 1 2

NB. Binary # (replicate)
NB. Left-operand acts as a mask that indicates how many times the element is replicated.
0 1 1 0 # _2 _3 2 3
0 1 2 3 # 0 1 2 3
1 0 1 # ~ 1 2 3

NB. Both sides must have the same size except when one of them has just one element.


NB. Unary # to binary #
NB. Applying ~ to a unary #, turns it into a binary # where the second operand is generated by refelexivity.
#~ 1 2 3

NB. Function Composition.
sum =: +:
sq_sum =: sum @: square
sq_sum 1 2 3

mod2 =: 2 | ]
eq0 =: 0 = ]
pair =: eq0 @: mod2
pair i. 6

inc =: 1 + ]
test =: +/ @: inc @: i.
test 3
test