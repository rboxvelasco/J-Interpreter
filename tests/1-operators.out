3
_3
3
1
0 2 4 5
4 5 0 2
0 1 2 3 4
2 1 0
8 7 4
2 4 6
1 4 9
6
30
3 2 1
3
_3 2
1 2 2 3 3 3
1 0 0 1 1 1
1 2 2 3 3 3
2 8 18
1 0 1 0 1 0
6
+/ @: 1 + ] @: i.
